### PR TITLE
RESTEASY-1341 - #792 rebased against current master

### PR DESCRIPTION
**Pull request by asoldano**

### CODE REVIEW

**Comment by asoldano** [comment]
@NicoNes can you have a look at the failures?

**Comment by NicoNes** [comment]
Hi @asoldano,

I started to watch the failures and it seems that I've forgot to add some `uriBuilder.clone` in methods of `ClientWebTarget`.
I checked out your branch but I did not succeed in running tests of `resteasy-jaxrs-testsuite` module. Tests seem to failed because of error such as : 
> Running org.jboss.resteasy.test.nextgen.encoding.EncodingTest
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 1.825 sec <<< FAILURE! - in org.jboss.resteasy.test.nextgen.encoding.EncodingTest
org.jboss.resteasy.test.nextgen.encoding.EncodingTest  Time elapsed: 1.81 sec  <<< ERROR!
java.lang.ExceptionInInitializerError: null
        at org.jboss.logging.Logger$1.run(Logger.java:2556)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.jboss.logging.Logger.getMessageLogger(Logger.java:2529)
        at org.jboss.logging.Logger.getMessageLogger(Logger.java:2516)
        at org.jboss.resteasy.resteasy_jaxrs.i18n.LogMessages.<clinit>(LogMessages.java:26)
        at org.jboss.resteasy.plugins.providers.RegisterBuiltin.registerProviders(RegisterBuiltin.java:75)
        at org.jboss.resteasy.plugins.providers.RegisterBuiltin.register(RegisterBuiltin.java:31)
        at org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder.getProviderFactory(ResteasyClientBuilder.java:360)
        at org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder.build(ResteasyClientBuilder.java:396)
        at org.jboss.resteasy.test.nextgen.encoding.EncodingTest.setupContainer(EncodingTest.java:37)
org.jboss.resteasy.test.nextgen.encoding.EncodingTest  Time elapsed: 1.825 sec  <<< ERROR!
java.lang.NullPointerException: null
        at org.jboss.resteasy.test.nextgen.encoding.EncodingTest.teardownContainer(EncodingTest.java:45)

I'll try to make them work but If you have any idea why please let me know...
Anyway please find enclosed the updates version of ClientWebTarget  
[ClientWebTarget.txt](https://github.com/resteasy/Resteasy/files/453517/ClientWebTarget.txt), it will help fixing few tests.

File: resteasy-jaxrs-testsuite/src/test/java/org/jboss/resteasy/test/client/ClientInvocationBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
+package org.jboss.resteasy.test.internal;
```
Comment (NicoNes) [inline resteasy-jaxrs-testsuite/src/test/java/org/jboss/resteasy/test/client/ClientInvocationBuilderTest.java]: Wrong package declaration

**Comment by asoldano** [comment]
hi @NicoNes , I've rebased your commits again against current master, as there were quite a lot more changes / refactoring. I suggest to try building the whole project (mvn clean install from prj root), you should get first failure in Jettison provider module.
The tests in resteasy-jaxrs-testsuite have been migrated to testsuite/integration-tests.
As for the wrong package, you're right, but actually that new test would also need to be properly migrated to testsuite/integration-tests.

**Comment by NicoNes** [comment]
Hi @asoldano 

Thanks you :+1:  I succeeded in running the tests, I also migrated the one I created. New PR is https://github.com/resteasy/Resteasy/pull/937.

**Comment by asoldano** [comment]
OK, thanks @NicoNes . I'm closing this PR and having the new one reviewed also by Ron!

**Pull request closed without merging**